pkg/controller/daemon/daemoncontroller_test.go

```diff
@@ -763,6 +791,17 @@ func TestBadNameDaemonSetDoesNothing(t *testing.T) {
 	syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0)
 }
 
+// DaemonSet with node selector, matching a node, should launch a pod on the node.
+func TestSelectorDaemonSetLaunchesPods(t *testing.T) {
```

Comment:
There are many different cases for node selector, you can check https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemoncontroller_test.go#L766-L788 , I was trying to extend the cases a bit by adding one more case for node selector, this case will only cover the node selector. This test case does not related much to this PR, perhaps I can just move this out to a new PR?